torch/distributed/tensor/parallel/fsdp.py

```diff
@@ -270,9 +270,12 @@ def _chunk_dtensor(
         # For DTensors, it is sharded across tp dimension first and then sharded across FSDP dimension.
         # TP is the inner dimension and FSDP is the outer dimension.
         # Therefore, shard placements for tensor is (Shard(0), tp_placement).
+        # For higher dimensional meshes, it is replicated across other dimensions. For example, with
+        # HSDP the shard placements for tensor is (Replicate, Shard(0), tp_placement).
         replicate_placements = [Replicate() for _ in range(parent_mesh.ndim)]
         replicate_placements[-1] = tp_placement  # type: ignore[call-overload]
-        shard_placements = [DShard(0) for _ in range(parent_mesh.ndim)]  # type: ignore[misc]
+        shard_placements = [Replicate() for i in range(parent_mesh.ndim)]  # type: ignore[misc]
+        shard_placements[-2] = DShard(0)  # type: ignore[call-overload]
```

Comment:
not from this PR but I wonder why we aliased `Shard` as `DShard` haha (I have not seen that elsewhere)

Reply:
@awgu  I can rename it if you prefer?

Reply:
We are also importing `Shard` from ShardedTensor, so I had to come up with `DShard` haha.
https://github.com/pytorch/pytorch/blob/main/torch/distributed/tensor/parallel/fsdp.py#L10

Reply:
No need to rename! I was just curious. Thanks @wz337 for the explanation :)

Reply:
I dislike DShard lol, we should rename ShardedTensor to someelse and preserve Shard as we are moving away from ShardedTensor